### ISSUE TITLE
RoadRunner crashes with Biomodels 9

### ISSUE THREAD

**Issue by hsauro** [sys-bio/roadrunner#114]
It has been reported that CVODE in libRoadRunner crashes with Biomodels 09. Contact Kyle for details.

**Comment by 0u812** [comment]
The real problem with this case is that we don't set a *minimum* time step.

**Comment by 0u812** [comment]
I can get this to simulate but I can't get it to agree with COPASI. This model is not well-behaved, and has all sorts of numerical issues. I will take another pass later, but I'm not hopeful.

**Comment by hsauro** [comment]
I assume you are playing around with covde settings. Note also that COPASI
uses lsoda which is a different integrator.

Herbert

**Comment by 0u812** [comment]
RK4 version, works:
```
from roadrunner import RoadRunner, Logger

#Logger.setLevel(Logger.LOG_DEBUG) # too verbose
Logger.setLevel(Logger.LOG_WARNING)

# Load model
rr = RoadRunner('BIOMD0000000009.xml')

rr.setIntegrator('rk4')

#rr.getIntegrator().setValue('relative_tolerance', 1e-12)
#rr.getIntegrator().setValue('absolute_tolerance', 1e-12)

# Simulate model
results = rr.simulate(0, 50, 50000)

print(results)

# Plot
rr.plot()
```

**Comment by hsauro** [comment]
Note rk4 has no tolerance settings.

H

**Comment by 0u812** [comment]
Right, I thought I commented those out in my post.

J Kyle Medley, PhD Student
Bioengineering/BPSD, Sauro Lab
University of Washington, Seattle
medleyj@uw.edu

**Comment by 0u812** [comment]
I can get this to simulate using extremely small tolerances. Not sure why Jarnac can do it with looser tols, maybe it uses a stiff solver.

1e-14 (rel & abs):
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-1e-14.png)

1e-13:
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-1e-13.png)

5e-13:
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-5e-13.png)

**Comment by hsauro** [comment]
Jarnac like most decent modeling packages will default to a stiff solver by
default. libRoadRunner SHOUD default to stiff=true. I'll add this to github.


Herbert

**Comment by hsauro** [comment]
There something not quite right if you have to go down to 1E-14, is stiff =
true?

Herbert

**Comment by 0u812** [comment]
No, working on the stiff problem right now.

**Comment by 0u812** [comment]
Ew, stiff output looks worse:

1e-14:
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-stiff-1e-14.png)

1e-13:
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-stiff-1e-13.png)

5e-13:
![](https://raw.githubusercontent.com/wiki/sys-bio/roadrunner/img/biomod9/cvode-stiff-5e-13.png)

**Comment by 0u812** [comment]
Let me double check if all the pertinent parameters are being passed to the stiff solver.

**Comment by 0u812** [comment]
Fixed 1.3.22
